[doc] add a doc about adoption metrics (#2612)
sorbet/sorbet: website/docs/tracking_adoption.md

---
id: tracking_adoption
title: Measuring Type-checking Coverage
sidebar_label: Tracking Coverage
---

> There are two primary metrics that we recommend to track the adoption at Stripe:

Reviewer: ```suggestion
There are two primary metrics that we recommend to track adoption of Sorbet in a codebase:
```

> - File-level coverage: the number of files in each type

Reviewer: ```suggestion
- File-level coverage: the number of files at each
```

  [strictness level](static.md). There are 5 strictness levels
- Input send coverage: the number of method calls called on a typed object. Only
  reported in type-checked files (strictness level true or higher)

Together, they represent how much of a codebase is type-checked by sorbet.

```

> $ srb tc --metrics-file test.txt --metrics-prefix example

Reviewer: ```suggestion
$ srb tc --metrics-file metrics.json --metrics-prefix example
```

  {
   "name": "example.types.input.files.sigil.false",
   "value": 1216
  },
  {
   "name": "example.types.input.files.sigil.true",
   "value": 1660
  },
  {
   "name": "example.types.input.files.sigil.strict",
   "value": 334
  },
  ...
  {
   "name": "example.types.input.sends.typed",
   "value": 69921
  },
  {
   "name": "example.types.input.sends.total",
   "value": 120061
  },

> }

Reviewer: Do you need to `cat test.txt` for this to show up as output?

```

The metrics are important at different phases of adoption. During the ramp-up,

> the first metric is more important. During this phase, we should focus on

Reviewer: Might be worth repeating that "the first metric" means "file-level typedness"
